cinema/views.py

```diff
@@ -97,12 +91,27 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
-        if self.action == "retrieve":
+        elif self.action == "retrieve":
             return MovieDetailSerializer
-
+        elif self.action == "upload_image":
+            return MovieImageSerializer
         return MovieSerializer
 
+    @action(methods=["Post"],
+            detail=True,
+            url_path="upload_image")
+    def upload_image(self, request, *args, **kwargs):
+        movie = self.get_object()
+        serializer = (self.get_serializer_class()
+                      (movie, data=request.data, partial=True))
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response(serializer.errors,
+                            status=status.HTTP_400_BAD_REQUEST
+                            )
```

Comment:
Im mniejsza liczba wcięć, tym czytelniejszy kod. Jeśli warunek kończy się słówkiem kluczowym `return` możemy uprościć kod do takiej postaci:
```suggestion
        if not serializer.is_valid():
            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)

        serializer.save()
        return Response(serializer.data, status=status.HTTP_200_OK) 
```